### DISCUSSION TITLE
Are `/usr/bin/env python` lines needed?

### DISCUSSION THREAD

**Discussion by rominf** [pytest-dev/pytest-localserver#29] [General]
https://github.com/pytest-dev/pytest-localserver/blob/bebb98bc6fe220e459208c09134a934c314d12c9/pytest_localserver/plugin.py#L1 is useless since there is nothing except fixtures in the file.

I'm not sure about this:
https://github.com/pytest-dev/pytest-localserver/blob/bebb98bc6fe220e459208c09134a934c314d12c9/pytest_localserver/smtp.py#L1
since there is a code:
https://github.com/pytest-dev/pytest-localserver/blob/bebb98bc6fe220e459208c09134a934c314d12c9/pytest_localserver/smtp.py#L101-L122
but it's ancient; not sure if someone still uses it.

**Comment by diazona** [comment]
Agreed, it would also be my preference to remove that code. I don't think these files need to be runnable from the command line; or if we do want to allow that, it makes sense to put it in a console script. But I don't want to make a mess of @redtoad's fine work either :smile: and I don't know if maybe there _is_ some reason to keep that code around that I'm not seeing.

Anyway, if neither of my co-maintainers suggests otherwise, I'd be happy to accept a pull request that removes these parts of the code, if you feel like preparing one. Otherwise, I'll get around to it at some point (though likely not for the 0.6 release).

**Comment by redtoad** [comment]
Hmm... What was I thinking back then.

The shebang directive in `plugin.py` is useless, I agree. 

The one in `smtp.py`is not strictly necessary since this can be started using `python -m pytest_localserver.smtp`. The main block I added for testing, if I remember correctly, and would not remove.

**Reply by RonnyPfannschmidt** [reply]
i would recommend to move the main block to a main function and just calling it in the dunder main block

**Reply by diazona** [reply]
Could we put the testing code in a separate module? I kind of figured that's what we have test files for, e.g. `tests/test_smtp.py` or another test file.

**Reply by diazona** [reply]
I just realized what you meant by "testing" in this case, i.e. not _automated_ testing :facepalm: so yeah, it makes sense to keep the code but I would also put it in a function. I'll do that in a PR soon.

**Reply by diazona** [reply]
#67
